qiskit/passmanager/flow_controllers.py

```diff
@@ -32,7 +34,21 @@ class FlowController:
 
     registered_controllers = OrderedDict()
 
-    def __init__(self, passes, options, **partial_controller):
+    def __init__(
+        self,
+        passes: Sequence[GenericPass | "FlowController"],
+        options: dict,
+        **partial_controller: Callable,
+    ):
+        """Create new flow controller.
+
+        Args:
+            passes: passes to add to the flow controller. This must be normalized.
+            options: PassManager options.
+            **partial_controller: Partially evaluated callables representing
+                a condition to invoke flow controllers. This dictionary
+                must be keyed on the registered controller name.
+        """
         self._passes = passes
         self.passes = FlowController.controller_factory(passes, options, **partial_controller)
```

Comment:
I agree. I want to drop `controller_factory` (it becomes much readable though), and I don't think appending passes to pass manager with kwargs, e.g. .`append(passes, condition=callable1, do_while=callable2)`, is really good interface as it introduces implicit and uncontrollable pass hierarchy through the `FlowController` class variable. User must explicitly instantiate controllers rather than relying on this uncontrollable factory method.